src/modules/Home/components/MainSlider/components/MainSlide/MainSlide.tsx

```diff
@@ -0,0 +1,179 @@
+import cn from 'classnames';
+import React, {PureComponent} from 'react';
+
+import Terminal from 'assets/images/terminalWindow.svg';
+import Flex from 'common/components/Flex/Flex';
+import styles from './mainSlide.scss';
+
+interface IProps {
+	currentSlideIndex: number;
+	changeSlide: any;
+	title: string;
+	mainSlidesLength: number;
+	smallText: any;
+	order: number;
+	isAnimated: boolean;
+	className: 'string';
+	mainSlides: any[];
+}
+
+interface IState {
+	isMenuOpen: boolean;
+}
+
+class MainSlide extends PureComponent<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+		this.state = {
+			isMenuOpen: false,
+		};
+	}
+
+	public componentDidUpdate(props: IProps) {
+		if (props.order !== this.props.order) {
+			this.setState({isMenuOpen: false});
+		}
+	}
+
+	public toggleRightSideBar = () => {
+		this.setState(state => ({
+			isMenuOpen: !state.isMenuOpen,
+		}));
+	}
+
+	public render() {
+		const {
+			title,
+			smallText,
+			mainSlidesLength,
+			currentSlideIndex,
+			order,
+			isAnimated,
+			className,
+			mainSlides,
+		} = this.props;
+
+		const {isMenuOpen} = this.state;
+
+		const pagination = [];
+		for (let i = 0; i < mainSlidesLength; i++) {
+			pagination.push((
+				<div
+					key={i}
+					className={cn(
+						styles.slidePaginationPoint, {
+						[styles.slidePaginationPoint_active]: i === currentSlideIndex,
+					})}
+				/>
+			));
+		}
+
+		const rightSideMenuList: any = [];
+		for (const slide of mainSlides) {
+			rightSideMenuList.push((
+				<div
+					key={slide.title}
+					className={cn(styles.slideRightBarListItem, {
+						[styles.slideRightBarListItem_disabled]: mainSlides.indexOf(slide) === currentSlideIndex,
+					})}
+					onClick={() => this.props.changeSlide(mainSlides.indexOf(slide))}
+				>
+					{slide.title}
+				</div>
+			));
+		}
+
+		return (
+			<div
+				className={cn(styles.slide, {
+					[styles.slide_top]: order === 0,
+					[styles.slide_middle]: order > 0 && order < mainSlidesLength - 1,
+					[styles.slide_bottom]: order === mainSlidesLength - 1,
+					[styles.animated]: isAnimated,
+				})}
+				style={{zIndex: -order + 20}}
```

Comment:
what the crutch here?

Reply:
The logic here is: slides are absolutely positioned. Each have its 'order'. The slide with order '0' is above, 1 is below it and so on. So z-index should decrease while order increase. So that expression  doing numbers flip and making sure that our z-index'es will be above 0 by '+ 20'